spec/index.bs

```diff
@@ -3835,6 +3835,153 @@ dictionary GPURenderPipelineDescriptor : GPUPipelineDescriptorBase {
+            Otherwise:
+                - initialize the |vertexList| with |drawCall|.vertexCount integers
+                - assign the |vertexList| item |i| to be |drawCall|.firstVertex + |i|
+
+            Note: in case of indirect draw calls, the `indexCount`, `vertexCount`,
+            and other properties of |drawCall| are read from the indirect buffer
+            instead of the draw command itself.
+
+            Issue: specify indirect commands better.
+
+        1. Process vertices. Each vertex |vertexIndex| in the |vertexList|,
+            in each instance of index |rawInstanceIndex|, is processed independenty.
+            The |rawInstanceIndex| is in range from 0 to |drawCall|.instanceCount - 1, inclusive.
+            This |processing happens in parallel, and any side effects, such as
+            writes into {{GPUBufferBindingType/"storage"|GPUBufferBindingType."storage"}} bindings,
+            may happen in any order.
+            1. Let |instanceIndex| be |rawInstanceIndex| + |drawCall|.baseInstance.
+            1. For each non-`null` |vertexBufferLayout| under index |i|
+                in the list of |descriptor|.{{GPURenderPipelineDescriptor/vertex}}.{{GPUVertexState/buffers}}:
+                1. Let |vertexBuffer| and |vertexBufferOffset| be the buffer and offset in |vertexBuffers| bindings at slot |i|
+                1. Let |vertexElementIndex| be dependent on |vertexBufferLayout|.{{GPUVertexBufferLayout/stepMode}}:
+                    <dl class="switch">
+                        : {{GPUInputStepMode/"vertex"}}
+                        :: |vertexIndex|
+                        : {{GPUInputStepMode/"instance"}}
+                        :: |instanceIndex|
+                    </dl>
+                1. For each |attributeDesc| in |vertexBufferLayout|.{{GPUVertexBufferLayout/attributes}}:
+                    1. Let |attributeOffset| be |vertexBufferOffset| +
+                        |vertexElementIndex| * |vertexBufferLayout|.{{GPUVertexBufferLayout/arrayStride}} +
+                        |attributeDesc|.{{GPUVertexAttribute/offset}}.
+                    1. Load the attribute |data| of format |attributeDesc|.{{GPUVertexAttribute/format}}
+                        from |vertexBuffer| starting at offset |attributeOffset|.
+                    1. Convert the |data| into a shader-visible format.
+                        <div class="example">
+                        An attribute of type {{GPUVertexFormat/"uchar2norm"}} will be converted from
+                            2 bytes of fixed-point unsigned 8-bit integers into 2 floating-point values
+                            as `vec2<f32>` in WGSL.
+                        </div>
+                    1. Bind the |data| to vertex shader input
+                        location |attributeDesc|.{{GPUVertexAttribute/shaderLocation}}.
+            1. For each {{GPUBindGroup}} group at |index| in |bindGroups|:
```

Comment:
Perhaps an inner loop over the group members instead of "binding" the whole group?